src/script/common/c_packer.cpp

```diff
@@ -468,6 +479,16 @@ void script_unpack(lua_State *L, PackedValue *pv)
 				break;
 			case LUA_TTABLE:
 				lua_createtable(L, i.uidata1, i.uidata2);
+				lua_getglobal(L, "core");
+				lua_getfield(L, -1, "registered_async_metatables");
```

Comment:
small problem: the packer is also used to transfer data into the mapgen env, trying to read this variable there is a logic mistake

possible solutions:
1. make the packer aware of which environment it is running in
2. rename `registered_async_metatables` to something more generic like `known_metatables`

I prefer 2.

Also to keep everything generic the code shouldn't assume `core` to exist IMO